### ISSUE TITLE
Gallery app would like a way to report crashes

### ISSUE THREAD

**Issue by sethladd** [flutter/flutter#4070]
We'd like to capture and measure crashes in the wild. We should find a good way to do this in Flutter apps.

Perhaps we can try https://firebase.google.com/docs/crash/

**Comment by sethladd** [comment]
Implementing this might provide insight for https://github.com/flutter/flutter/issues/614

**Comment by sethladd** [comment]
cc @collinjackson who was interested in trying out Firebase Crash Reporting + Flutter

**Comment by eseidelGoogle** [comment]
Moving this to Gallery-Next since we have 2 other customers asking for this and should pave the way for them with the Gallery.

**Comment by rmacnak-google** [comment]
(Assuming we mean a C crash and not just an unhandled Dart exception.)

Currently precompilation supports two ways of outputting instructions, either outputting an assembly file that can be turned into a shared library that can be loaded with dlopen or outputting a binary blob that can be loaded with mmap. On iOS, Flutter uses the shared library approach (mmap can't create executable pages, which enforces the no-JITing policy) and as a benefit the native debugging tools have symbols for the precompiled Dart code. On Android, the binary blob approach is used to avoid a build step that requires the NDK on developer machines, but this means there are no symbols for precompiled Dart code. Getting symbols for precompiled code on Android might mean switching to shared libraries, or it might mean we need to output some kind of symbol blob as well, though what format would depending on what tool would be symbolizing the traces.

**Comment by collinjackson** [comment]
I don't think we are implementing Firebase Crash (Google is recommending NOT to use Firebase for crash reporting on new projects anymore) and adding Crashlytics is covered by #9592. So I think this issue is just about adding crash reporting to the Gallery, possibly with [package:sentry](https://pub.dartlang.org/packages/sentry).

Unassigned myself since I'm not currently working on that.

**Comment by Hixie** [comment]
cc @yjbanov is this done?

**Comment by Hixie** [comment]
cc @sethladd is this done?

**Comment by sethladd** [comment]
Not done. Would be perfect for a DPE. I'd like for a DPE to help us "dogfood" our crash reporting, and provide a sample/example.

**Comment by Hixie** [comment]
Ideally we'd just use whatever we tell other people to use, once we have an answer to that.

**Comment by eseidelGoogle** [comment]
Right now the best we have is probably package:sentry with https://github.com/flutter/crashy as an example of using it.  Once we have a more official Crashlytics package that would be a good one to use here: https://github.com/flutter/flutter/issues/14765

**Comment by yjbanov** [comment]
We may also be able to use the same thing we use for flutter command-line tools (see [crash_reporting.dart](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/crash_reporting.dart)).

**Comment by amanv8060** [comment]
Can we close this in favor of https://github.com/flutter/gallery/issues/632 ?

**Comment by amanv8060** [comment]
cc/ @guidezpl

**Comment by guidezpl** [comment]
This has more history, so let me move it to the repo, and use this for tracking
